ortools/sat/csharp/CpModel.cs

```diff
@@ -615,6 +615,14 @@ public void AddHint(IntVar var, long value)
             model_.SolutionHint.Values.Add(value);
         }
 
+        public void AddAssumptions(IEnumerable<ILiteral> literals)
+        {
+            foreach (ILiteral lit in literals)
+            {
+                model_.Assumptions.Add(lit.GetIndex());
+            }
+        }
+
```

Comment:
add single var API, checks Positivity: 
if (lit.GetIndex() >= 0) {
  model_.Assumptions.Add(lit.GetIndex());
} else {
  model_.Assumptions.Add(lit.Not().GetIndex());
}

to be done in python, java, and .NET

Reply:
Doesn't this change the semantics for a (nonsensical) case like this?
```python
x = model.NewIntVar(0, 10, "")
a = model.NewBoolVar("")

model.Add(x > 10).OnlyEnforceIf(a.Not())
model.Proto().assumptions.append(a.Not().Index())
```
How about raise an error instead?